src/test/ui/proc-macro/allowed-attr-stmt-expr.stdout

```diff
@@ -0,0 +1,309 @@
+                ],
+                span: $DIR/allowed-attr-stmt-expr.rs:52:24: 52:42 (#0),
+            },
+            Ident {
+                ident: "struct",
+                span: $DIR/allowed-attr-stmt-expr.rs:52:43: 52:49 (#0),
+            },
+            Ident {
+                ident: "Bar",
+                span: $DIR/allowed-attr-stmt-expr.rs:52:50: 52:53 (#0),
+            },
+            Group {
+                delimiter: Brace,
+                stream: TokenStream [],
+                span: $DIR/allowed-attr-stmt-expr.rs:52:54: 52:56 (#0),
+            },
+        ],
+        span: $DIR/allowed-attr-stmt-expr.rs:52:22: 52:57 (#0),
+    },
+    Punct {
+        ch: ';',
+        spacing: Alone,
+        span: $DIR/allowed-attr-stmt-expr.rs:52:57: 52:58 (#0),
+    },
+]
+PRINT-ATTR INPUT (DISPLAY): #[allow(unused)] #[allow(dead_code)] struct Bar { }
+PRINT-ATTR INPUT (DEBUG): TokenStream [
+    Punct {
+        ch: '#',
+        spacing: Alone,
+        span: $DIR/allowed-attr-stmt-expr.rs:28:9: 28:10 (#34),
+    },
+    Group {
+        delimiter: Bracket,
+        stream: TokenStream [
+            Ident {
+                ident: "allow",
+                span: $DIR/allowed-attr-stmt-expr.rs:28:11: 28:16 (#34),
+            },
+            Group {
+                delimiter: Parenthesis,
+                stream: TokenStream [
+                    Ident {
+                        ident: "unused",
+                        span: $DIR/allowed-attr-stmt-expr.rs:28:17: 28:23 (#34),
+                    },
+                ],
+                span: $DIR/allowed-attr-stmt-expr.rs:28:16: 28:24 (#34),
+            },
+        ],
+        span: $DIR/allowed-attr-stmt-expr.rs:28:10: 28:25 (#34),
+    },
+    Punct {
+        ch: '#',
+        spacing: Alone,
+        span: $DIR/allowed-attr-stmt-expr.rs:52:23: 52:24 (#0),
+    },
+    Group {
+        delimiter: Bracket,
+        stream: TokenStream [
+            Ident {
+                ident: "allow",
+                span: $DIR/allowed-attr-stmt-expr.rs:52:25: 52:30 (#0),
+            },
+            Group {
+                delimiter: Parenthesis,
+                stream: TokenStream [
+                    Ident {
+                        ident: "dead_code",
+                        span: $DIR/allowed-attr-stmt-expr.rs:52:31: 52:40 (#0),
+                    },
+                ],
+                span: $DIR/allowed-attr-stmt-expr.rs:52:30: 52:41 (#0),
+            },
+        ],
+        span: $DIR/allowed-attr-stmt-expr.rs:52:24: 52:42 (#0),
+    },
+    Ident {
+        ident: "struct",
+        span: $DIR/allowed-attr-stmt-expr.rs:52:43: 52:49 (#0),
+    },
+    Ident {
+        ident: "Bar",
+        span: $DIR/allowed-attr-stmt-expr.rs:52:50: 52:53 (#0),
+    },
+    Group {
+        delimiter: Brace,
+        stream: TokenStream [],
+        span: $DIR/allowed-attr-stmt-expr.rs:52:54: 52:56 (#0),
+    },
+]
+PRINT-ATTR INPUT (DISPLAY): #[allow(unused)] struct Other { } ;
```

Comment:
It looks like we were previously treating the semicolon as part of the item, but this wasn't visible to macros since we invoked the attribute on the item (not the statement).
I think treating the semicolon as a separate empty statement is the correct behavior. I'll work on a fix.